vcpkg/consuming-packages/configure-git-authentication-in-github-actions.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Configure Git authentication in GitHub Actions
+description: Learn to configure Git credentials for private repositories when using vcpkg on GitHub Actions
+author: vicroms
+ms.author: viromer
+ms.date: 7/20/2023
+ms.prod: vcpkg
+---
+# Configure Git authentication in GitHub Actions
+
+When using vcpkg with private registries it is necessary to provide access
+credentials to authenticate Git operations executed internally by vcpkg.
+
+In this article we will show you how to authenticate to a private vcpkg registry
+hosted on GitHub to use in your GitHub Actions workflows.
+
+## Step 1: Create a Personal Access Token for your private repository
+
+The first step is to create a Personal Access Token (PAT) for your GitHub account.
+
+- Log into GitHub using an account with access to the private repository.
+- Go to <https://github.com/settings>
+- Click on _Developer Settings_
+- Click on _Tokens (classic)_
+- Click the _Generate new token (classic)_
+- Give your token appropiate description, expiration date, and permissions. Make
+  sure that the `repo` permission is enabled, this will give the PAT full access
+  to your private repositories.
+- Click on _Generate token_
+- Copy and paste the generated PAT in a secure location, you won't be able to
+  view it again after leaving the current page
+- (Optional) If your organization uses SSO, you need to enable it for the PAT
```

Comment:
I hesitate to reproduce detailed steps like these in our documentation when GitHub's documentation is likely to be less brittle when they make changes in the future. Especially so when it comes to authentication/authorization features.

Maybe https://docs.github.com/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens

Reply:
> Is it possible/meaningful to generalize this to work for other hosts of private registries (such as an AzDO hosted registry).

We can add a note that the steps here should work with any Git registry regardless of the hosting service. I chose to use GitHub specifically to have an end-to-end scenario.

I agree with removing the steps to get a PAT from this article in favor of linking to GitHub's documentation.